elf/input-files.cc

```diff
@@ -740,6 +746,35 @@ void ObjectFile<E>::register_section_pieces(Context<E> &ctx) {
   }
 }
 
+template <typename E>
+void ObjectFile<E>::fill_addrsig(Context<E> &ctx) {
+  if (llvm_addrsig) {
+    const u8 *start = reinterpret_cast<const u8 *>(llvm_addrsig->contents.data());
+    const u8 *end = start + llvm_addrsig->contents.size();
+    const u8 *cur = start;
+    while (cur != end) {
+      u64 symIndex = read_uleb(cur);
+      if (this->symbols[symIndex]->file != this) continue;
```

Comment:
I added checks so it will prevent any potential race condition, but I didn't manage to understand why a symbol could be from another file's section. What are the cases where this happen?